src/finch_tracker/finch_tracker_main.ts

```diff
@@ -0,0 +1,185 @@
+    .description('Chrome finch tracker')
+    .version('0.0.1')
+    .argument('<finch_storage>', 'A path to a repository to store Finch data')
+    .argument(
+      '[current_seed_file]',
+      'Explicitly set the current seed file.' +
+        'By default the file is downloaded from the backend.',
+    )
+    .argument(
+      '[previous_seed_file]',
+      'Explicitly set the previous seed file.' +
+        '<finch_storage>/seed.bin is used by default.',
+    )
+    .option(
+      '-m, --chrome-major <number>',
+      'Override the current stable major chrome version.' +
+        'By default the version is taken from the backend' +
+        '(see kGetUsedChromiumVersion)',
+      parseInt,
+    )
+    .option('--no-update', "Don't make any disk changes")
+    .option('--no-commit', "Just update <finch_storage>, don't create a commit")
+    .option('-o --output <file>', 'A file to create a summary')
+    .parse();
+
+  const storageDir = program.args[0];
+  const seedFile = program.args[1];
+  const previousSeedFile = program.args[2];
+  let minMajorVersion = program.opts().chromeMajor;
+
+  if (minMajorVersion === undefined) {
+    const chromiumVersionData = await downloadUrl(kGetUsedChromiumVersion);
+    const chromiumVersionString = chromiumVersionData?.toString().split('.')[0];
+    if (chromiumVersionString === undefined) {
+      program.error(
+        'Failed to get the Chromium version via ' + kGetUsedChromiumVersion,
+      );
+      return;
+    }
+    console.log('Got Chromium version', chromiumVersionString);
+    minMajorVersion = parseInt(chromiumVersionString);
+  }
+
+  const options: ProcessingOptions = {
+    minMajorVersion,
+  };
+
+  const outputFile = program.opts().output;
+  const createSummary = outputFile !== undefined;
+  const updateData = program.opts().update;
+  const commitData = program.opts().commit && updateData;
+
+  if (!createSummary && !updateData) {
+    program.error('Nothing to do.');
+    return;
+  }
+
+  const seedData =
+    seedFile !== undefined
+      ? fs.readFileSync(seedFile)
+      : await fetchChromeSeedData();
+  const seed = proto.VariationsSeed.decode(seedData);
+  let previousSeedData: Buffer | undefined;
+  let newGitSha1: string | undefined;
+
+  if (createSummary) {
+    previousSeedData = fs.readFileSync(
+      previousSeedFile ?? getSeedPath(storageDir),
+    );
+  }
+
+  if (updateData) {
+    storeDataToDirectory(seedData, storageDir, options);
+    if (commitData) {
+      newGitSha1 = commitAllChanges(storageDir);
+      console.log('New commit sha1', newGitSha1);
+    }
+  }
+
+  if (createSummary && previousSeedData !== undefined) {
+    const previousSeed = proto.VariationsSeed.decode(previousSeedData);
+    const summary = makeSummary(
+      previousSeed,
+      seed,
+      options,
+      StudyPriority.STABLE_MIN,
+    );
+    const summaryJSON = summaryToJson(summary, newGitSha1);
+    fs.writeFileSync(outputFile, summaryJSON);
+  }
+}
+
+void main();
```

Comment:
not sure why we have this `void` part here. You might want to handle any exception, something like:
```
main()
    .then(() => process.exit(0))
    .catch(err => {
        console.error('Error running command:')
        console.error(err)
        process.exit(1)
    })
```

If there's any specific status you want to give. Not sure if node is showing you some kind of log about a dangling unhandled promise without that part?

Reply:
Thanks, fixed.